main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -768,6 +831,10 @@ object Lowering {
         case _ => tpe0
       }
 
+      // Special case for Channel[_], which is rewritten to Concurrent/Channel.Mpmc
+      case Type.Cst(TypeConstructor.Channel, loc) =>
```

Comment:
Looks like we should add `@EliminatedBy(Lowering.class)` annotation to `TypeConstructor.Channel`.

Reply:
OK, sure

Reply:
Some of us still believe in that annotation :D